src/components/Article.js

```diff
@@ -2,24 +2,45 @@ import { useEffect, useState } from "react";
 import { Link, useParams } from "react-router-dom";
 import { fetchArticleById } from "../api";
 import ArticleTitleCard from "./ArticleTitleCard";
+import Comments from "./Comments";
 
 export default function Article() {
   const [currentArticle, setCurrentArticle] = useState({});
   const { article_id } = useParams();
+  const [isLoading, setLoading] = useState(true)
+  const [viewCommentsString,setViewCommentStr]=useState('View Comments')
+  const [commentsComponent,setCommentsComponent]= useState('')
   useEffect(() => {
     fetchArticleById(article_id).then(({ article }) => {
       setCurrentArticle(article);
+      setLoading(false)
     });
   }, [currentArticle, article_id]);
+  function viewComments(){
+    setLoading(true)
+    if(viewCommentsString==='View Comments'){
+      setViewCommentStr('Hide Comments')
+      setCommentsComponent(<Comments/>)
+      setLoading(false)
+    }
+    if(viewCommentsString==='Hide Comments'){
+      setViewCommentStr('View Comments')
+      setCommentsComponent('')
+      setLoading(false)
+    }
+  }
```

Comment:
this works but you can use in-line conditional rendering to tidy this up

(imagine you have some state called `showComments` which is a boolean here...)

```diff
-      <button onClick={viewComments}> {viewCommentsString}</button>
+      <button onClick={viewComments}>{showComments ? "Hide Comments" : "Show Comments"}</button>
```

```diff
-    {commentsComponent}
+    {showComments ? <Comments /> : null}  
```

Reply:
check the conditional rendering notes!